src/main/java/com/iota/iri/service/TipsManager.java

```diff
@@ -161,7 +161,7 @@ Hash markovChainMonteCarlo(final Set<Hash> visitedHashes, final Map<Hash, Long>
                 monteCarloIntegrations.put(tail,1);
             }
         }
-        return monteCarloIntegrations.entrySet().stream().reduce((a, b) -> {
+        Hash resultTip = monteCarloIntegrations.entrySet().stream().reduce((a, b) -> {
```

Comment:
Did you leave out the case where a.getValue() == b.getValue() on purpose?

Reply:
`seed.nextBoolean()` a couple lines below it